apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,55 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
+import { ListItemTemplateDirective } from '../../directive/list-item-template.directive';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
-  templateUrl: './card.component.html',
+  template: `
+    <ng-content select="img"></ng-content>
+
+    <section>
+      <ng-container
+        *ngFor="let item of list; trackBy: trackById"
+        [ngTemplateOutlet]="listItemTemplate"
+        [ngTemplateOutletContext]="{ $implicit: item }">
+      </ng-container>
+    </section>
+
+    <button
+      class="border border-blue-500 bg-blue-300 p-2 rounded-sm"
+      (click)="addNewItem()">
+      Add
+    </button>
+  `,
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgFor, ListItemComponent, NgTemplateOutlet],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
-
-  CardType = CardType;
+export class CardComponent<
+  ListItem extends { id: number },
+  T extends { $implicit: ListItem }
+> {
+  @ContentChild(ListItemTemplateDirective, { read: TemplateRef })
+  listItemTemplate!: TemplateRef<T>;
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
+  @Input() list!: ListItem[];
+  @Output() addItem = new EventEmitter<void>();
 
   addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
+    this.addItem.emit();
+  }
+
+  trackById(_index: number, item: ListItem) {
+    return item.id;
```

Comment:
in a few days, you will be able to use the new control flow with the new track function 👍 but nice to have implement it, not everyone think of it.